bignum.c

```diff
@@ -35,7 +35,7 @@ sexp sexp_fixnum_to_bignum (sexp ctx, sexp a) {
   return res;
 }
 
-sexp sexp_make_integer (sexp ctx, sexp_lsint_t x) {
+sexp sexp_make_integer (sexp ctx, sexp_sint_t x) {
```

Comment:
We should preserve the ability to make integers from long ints.

Reply:
It won't allow declaring `sexp_lsint_t` as a `struct` on platforms such as Visual Studio Win64 -- it doesn't support larger integer types that would satisfy `sizeof(sexp_sint_t) < sizeof(sexp_lsint_t)`. As C99 does not require any integer types that would larger than platform's word size, requiring it would sacrifice portability.

When will we need `sexp_make_integer` from long ints? In case we should interface with C APIs, I think we should have separate C type variants, say `sexp_make_integer_from_long` or `sexp_make_integer_from_intptr` because we still need to cast an integer from C API to `sexp_(l)sint_t` which has platform-defined size.

Single `sexp_make_integer` API would require knowledge on width of long ints:

```c
uintptr_t res = some_foreign_api_returns_ptr();
sexp out = (sizeof(sexp_lsint_t) >= sizeof(long)) ?
  sexp_make_integer(ctx, res) : "blame we might not have BIGNUM here";
```

Just go straight with `sexp_make_integer` would implicitly truncate C API result -- fortunately, it won't occur on current chibi-scheme supported platforms though.

Reply:
sexp_make_integer is used frequently, notably from all FFI calls involving integer types, where we may not even trivially know whether it's a long if it's a typedef like offset_t.

sexp_lsint_t is just a long long int, which C99 guarantees support for.  It doesn't guarantee to be larger than a long, but the alternative sexp_make_integer_from_long doesn't improve the situation.